source/_posts/angular-signals.md

```diff
@@ -0,0 +1,409 @@
+
+The important thing to remember is that each signal reading within `computed()` is *treated as a dependency*, and it will be tracked in the dependency graph.
+
+```ts
+const counter = signal(0);
+const isEqual = computed(() => counter() === 5); // counter() first dependency
+const isOdd = computed(() => counter() % 2); // counter() second dependency
+```
+
+And if we now change the signal value e.g. with `.set()`, the changes will be propagated in all `computed` functions, where the signal reading `counter()` is located. These locations are dependencies.
+
+{% img "ohh-yea.jpg" " " "lazy" %}
+
+It is worth mentioning that `computed()` values are cached. They are updated only when it’s needed.
+
+### Signal Effects
+
+`effect()` is a tree-shakable side effect function that **triggers when a signal value changes**. All signal dependencies are captured inside the function. Then, if there is a change in the signal, the side effect function is re-executed.
+
+```ts
+const counter = signal(0);
+
+effect(() => {
+    console.log('Counter value ', this.counter());
+
+    if (this.counter() === 5) {
+        console.log('Counter value is equal 5!');
+    }
+});
+
+
+counter.set(1);
+counter.set(2);
+counter.set(5);
+
+
+// Counter value 0
+// Counter value 1
+// Counter value 2
+// Counter value 5
+// Counter value is equal 5!
+```
+
+As you noticed, we were given the value `0` at the beginning. This is an initial value, given in the same way as `BehaviorSubject` in RxJS. Then we set the value 3 times and got the final message that the value is equal to `5`. 
+
+The Signal Effect behaves like a "listener" that needs changing dependencies (signals). Then this function is called. This is the perfect place to perform other operations, calling methods, etc.
+
+## Example of Angular Signals
+If you already have the minimal version of `16.0.0-next.0` installed, go ahead and copy the example and test how Angular Signals work.
+
+```ts
+import { Component, OnInit, computed, effect, signal } from '@angular/core';
+
+@Component({
+   selector: 'app-root',
+   template: `
+    <div>Count: {{ counter() }}</div>
+    <div>Odd: {{ isOdd() }}</div>
+    <div>Is equal to 5: {{ isEqual() }}</div>
+
+    <button (click)="increment()">Increment</button>
+    <button (click)="reset()">Reset</button>
+    <button (click)="decrement()">Decrement</button>
+  `,
+})
+export class AppComponent implements OnInit {
+   counter = signal(0);
+   isEqual = computed(() => this.counter() === 5);
+   isOdd = computed(() => this.counter() % 2);
+
+   ngOnInit() {
+      effect(() => {
+         console.log('Counter value', this.counter());
+
+         if (this.counter() === 5) {
+            console.log('Counter value is equal 5!');
+         }
+      });
+   }
+
+   increment() {
+      this.counter.update((value) => value + 1);
+   }
+
+   decrement() {
+      this.counter.update((value) => value - 1);
+   }
+
+   reset() {
+      this.counter.set(0);
+   }
+}
```

Comment:
Since when do developers use 3 spaces as indentation? I suggest using 2 spaces :)

Reply:
Webstorm IDE by default. Is this really important?